core/src/main/scala/com/avast/clients/rabbitmq/publisher/BaseRabbitMQProducer.scala

```diff
@@ -49,33 +54,40 @@ class DefaultRabbitMQProducer[F[_], A: ProductConverter](name: String,
     }
 
     converter.convert(body) match {
-      case Right(convertedBody) => send(routingKey, convertedBody, finalProperties)
+      case Right(convertedBody) =>
+        for {
+          _ <- checkSize(convertedBody, routingKey)
+          _ <- logErrors(sendMessage(routingKey, convertedBody, finalProperties), routingKey)
+        } yield ()
       case Left(ce) => Sync[F].raiseError(ce)
     }
   }
 
-  private def send(routingKey: String, body: Bytes, properties: MessageProperties)(implicit correlationId: CorrelationId): F[Unit] = {
-    checkSize(body, routingKey) >>
-      logger.debug(s"Sending message with ${body.size()} B to exchange $exchangeName with routing key '$routingKey' and $properties") >>
-      blocker
-        .delay {
-          sendLock.synchronized {
-            // see https://www.rabbitmq.com/api-guide.html#channel-threads
-            channel.basicPublish(exchangeName, routingKey, properties.asAMQP, body.toByteArray)
-          }
-        }
-        .flatTap(_ => sentMeter.mark)
-        .recoverWith {
-          case ce: AlreadyClosedException =>
-            logger.debug(ce)(s"[$name] Failed to send message with routing key '$routingKey' to exchange '$exchangeName'") >>
-              sentFailedMeter.mark >>
-              F.raiseError[Unit](ChannelNotRecoveredException("Channel closed, wait for recovery", ce))
-
-          case NonFatal(e) =>
-            logger.debug(e)(s"[$name] Failed to send message with routing key '$routingKey' to exchange '$exchangeName'") >>
-              sentFailedMeter.mark >>
-              F.raiseError[Unit](e)
+  protected def basicSend(routingKey: String, body: Bytes, properties: MessageProperties)(implicit correlationId: CorrelationId): F[Unit] = {
+    for {
+      _ <- logger.debug(s"Sending message with ${body.size()} B to exchange $exchangeName with routing key '$routingKey' and $properties")
+      _ <- blocker.delay {
+        sendLock.synchronized {
+          // see https://www.rabbitmq.com/api-guide.html#channel-threads
+          channel.basicPublish(exchangeName, routingKey, properties.asAMQP, body.toByteArray)
         }
+      }
+      _ <- sentMeter.mark
+    } yield ()
+  }
+
+  private def logErrors(from: F[Unit], routingKey: String)(implicit correlationId: CorrelationId): F[Unit] = {
```

Comment:
IMHO it shouldn't be named `log` because it doesn't do just logging (e.g. it's wrapping the error in one case).